tools/_setup_generation/step_knowedge_base/item.py

```diff
@@ -0,0 +1,131 @@
+import os
+from typing import TextIO, Dict, List
+
+from .exceptions import WrongHeader, NoHeader, NoIndexFile
+
+
+class Item:
+    CATEGORIES = ["tutorials", "demos", "tips"]
+    TYPES = ["code", "video", "article"]
+
+    def __init__(self, parent_path: str, file_path: str):
+        self.parent_path = parent_path  # ex: docs/knowledge_base/tips
+        self.file_path = file_path  # ex: docs/knowledge_base/tips/charts.md or docs/knowledge_base/tips/charts/index.md
+        if not os.path.exists(self.file_path):
+            raise NoIndexFile(f"File {self.file_path} not found")
+        with open(self.file_path) as file:
+            self.header = self._read_header(file)
+            self.title = self.header.get("title")
+            self.category = self.header.get("category", "tips")
+            self.type = self.header.get("type", "code")
+            self.data_keywords = self.header.get("data-keywords")
+            self.short_description = self.header.get("short-description")
+            self.img = self.header.get("img")
+        self._check_header()
+
+    @property
+    def href(self) -> str:
+        raise NotImplemented
+
+    @property
+    def icon(self) -> str:
+        """Return the icon path."""
+        return f"images/icon-{self.type}.svg"
+
+    def generate_content_for_tutorials(self) -> str:
```

Comment:
I would have used an 'indentation' parameter with the '    ' prefix so embedding is easier...